runtime/src/main/java/com/dylibso/chicory/runtime/ByteArrayMemory.java

```diff
@@ -139,7 +139,7 @@ public int waitOn(int address, int expected, long timeout) {
 
         synchronized (monitor) {
             try {
-                if (((int) INT_ARR_HANDLE.getVolatile(buffer, address)) == expected) {
```

Comment:
`getVolatile` throws on Zulu 25 as unsupported, but we are in a `synchronized` block so it might be excess of diligence using it here.